etc/notebooks/got_scotch_demo/scotch_dashboard_declarative.ipynb

```diff
@@ -0,0 +1,783 @@
+    "urth": {
+     "dashboard": {
+      "hidden": true
+     }
+    }
+   },
+   "source": [
+    "N.B. We're using absolute paths here to allow this notebook to work even when deployed as a dashboard where Thebe requests a kernel. When it does, it has the Jupyter notebook working directory as the `pwd` in which case a relative path will not work here. \n",
+    "\n",
+    "This is one of the challenges to be addressed with dashboards when deployed outside the authoring environment: access to data. A simple answer: put the data in a real data store, not local disk."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false,
+    "urth": {
+     "dashboard": {}
+    }
+   },
+   "outputs": [],
+   "source": [
+    "features_df = pd.read_pickle('/home/jovyan/work/got_scotch_demo/datasets/features.dataframe')\n",
+    "sim_df = pd.read_pickle('/home/jovyan/work/got_scotch_demo/datasets/sims.dataframe')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "collapsed": true,
+    "urth": {
+     "dashboard": {
+      "hidden": true
+     }
+    }
+   },
+   "source": [
+    "We will drop the cluster column from *features\\_df*. We don't need it here."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false,
+    "urth": {
+     "dashboard": {}
+    }
+   },
+   "outputs": [],
+   "source": [
+    "features_df = features_df.drop('cluster', axis=1)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "urth": {
+     "dashboard": {
+      "hidden": true
+     }
+    }
+   },
+   "source": [
+    "## Define Functions <span style=\"float: right; font-size: 0.5em\"><a href=\"#Got-Scotch?\">Top</a></span>"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "urth": {
+     "dashboard": {
+      "hidden": true
+     }
+    }
+   },
+   "source": [
+    "Now we need a couple of functions to retrieve similar scotches and get the feature table of a scotch.\n",
+    "\n",
+    "First, let's define a *get_similar* function, essentially a copy from the [scotch_dashboard notebook](./scotch_dashboard.ipynb) to return the data of the top n similar scotches to a given scotch."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true,
+    "urth": {
+     "dashboard": {}
+    }
+   },
+   "outputs": [],
+   "source": [
+    "def get_similar(name, n, top=True):\n",
+    "    a = sim_df[name].order(ascending=False)\n",
+    "    a.name = 'Similarity'\n",
+    "    df = pd.DataFrame(a) #.join(features_df).iloc[start:end]\n",
+    "    return df.head(n) if top else df.tail(n)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "urth": {
+     "dashboard": {
+      "hidden": true
+     }
+    }
+   },
+   "source": [
+    "But since urth-viz-table needs the index (distillery) as a regular column with the data, we need to add that column."
```

Comment:
Same thing about the second mention of adding a column. We're not adding a column but rather turning the DataFrame index into a column.